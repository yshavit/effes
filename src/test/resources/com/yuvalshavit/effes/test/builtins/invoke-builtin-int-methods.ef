test:
  print 33 . + 22

main:
  test 
