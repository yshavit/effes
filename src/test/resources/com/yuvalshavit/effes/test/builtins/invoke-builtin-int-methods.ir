def test () -> no result type:
  MethodInvoke
    method: print [built-in]
    arg0:
      MethodInvoke
        method: IntValue::+ [built-in]
        arg0:
          IntLiteral
            33
        arg1:
          IntLiteral
            22
  MethodInvoke
    method: print [built-in]
    arg0:
      MethodInvoke
        method: IntValue::- [built-in]
        arg0:
          IntLiteral
            33
        arg1:
          IntLiteral
            22
  MethodInvoke
    method: print [built-in]
    arg0:
      MethodInvoke
        method: IntValue::* [built-in]
        arg0:
          IntLiteral
            33
        arg1:
          IntLiteral
            10
  MethodInvoke
    method: print [built-in]
    arg0:
      MethodInvoke
        method: IntValue::/ [built-in]
        arg0:
          IntLiteral
            33
        arg1:
          IntLiteral
            10
