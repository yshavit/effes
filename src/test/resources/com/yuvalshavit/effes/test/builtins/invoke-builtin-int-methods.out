Your argument was: 55
>> null
